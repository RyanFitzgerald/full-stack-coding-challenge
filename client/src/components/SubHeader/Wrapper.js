import styled from 'styled-components';

// Import static files
import BG from '../../static/bg.png';

const Wrapper = styled.div`
  background: #f5f5f5;
  background-image: url(${BG});
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  padding: 100px 0;
  text-align: center;
`;

export default Wrapper;