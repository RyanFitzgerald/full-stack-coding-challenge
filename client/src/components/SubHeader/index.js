import React from 'react';
import PropTypes from 'prop-types';

// Import components
import Wrapper from './Wrapper';
import Container from '../Container';

// Define the component
function SubHeader({children}) {
  return (
    <Wrapper>
      <Container>
        {children}
      </Container>
    </Wrapper>
  );
}

// Define the props
SubHeader.propTypes = {
  children: PropTypes.element.isRequired
}

export default SubHeader;