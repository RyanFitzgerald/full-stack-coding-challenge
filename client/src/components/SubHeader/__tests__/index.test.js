import React from 'react';

import SubHeader from '../index';

describe('<SubHeader />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <SubHeader>Content</SubHeader>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its content', () => {
    const children = 'Some Content';
    const renderedComponent = shallow(
      <SubHeader>{children}</SubHeader>
    );
    
    expect(renderedComponent.contains(children)).toBe(true);
  });
});