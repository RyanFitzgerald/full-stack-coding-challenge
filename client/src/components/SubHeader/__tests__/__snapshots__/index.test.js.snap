// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubHeader /> should match the snapshot 1`] = `
.c0 {
  background: #f5f5f5;
  background-image: url(bg.png);
  border-bottom: 1px solid #e6e6e6;
  padding: 100px 0;
  text-align: center;
}

.c1 {
  margin: 0 auto;
  max-width: 960px;
  padding: 15px;
}

.c1:after {
  content: '';
  display: table;
  clear: both;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Content
  </div>
</div>
`;
